src/providers/database/FirebaseClient.ts

```diff
@@ -65,14 +71,53 @@ export class FirebaseClient implements IFirebaseClient {
       total
     };
   }
+  private async apiGetListLazy(
+    resourceName: string,
+    params: messageTypes.IParamsGetList
+  ): Promise<messageTypes.IResponseGetList> {
+    const r = await this.tryGetResource(resourceName);
+    log("apiGetListLazy", { resourceName, params });
+    let query: Query = r.collection;
+
+    const filters = this.options.softDelete ? {
+      ...params.filter,
+      deleted: false
+    } : params.filter;
+
+    query = this.filtersToQuery(query, filters);
+    query = this.sortToQuery(query, params.sort);
+
+    const { page, perPage } = params.pagination;
+    const fullParams = { ...params, filter: filters };
+
+
+    if (page === 1) {
+      query = query.limit(perPage);
+    } else {
+      let queryCursor = this.getQueryCursor(params);
+      if (!queryCursor) {
+        queryCursor = await this.findLastQueryCursor(query, fullParams);
+      }
+      query = query.startAfter(queryCursor).limit(perPage);
+    }
+
+    const snapshots = await query.get();
+    const data = snapshots.docs.map(doc => parseFireStoreDocument(doc));
+    this.setQueryCursor(
+      snapshots.docs[snapshots.docs.length - 1],
+      fullParams
+    );
+    return { data, total: 9999 };
```

Comment:
Out of curiosity...what's this 9999? Does it return the count of items to page through? If so should it be hard coded like this?

Reply:
Right, I should leave comment here. It's temporary, as it isn't possible to get count of documents without reading all of them in firebase. So I hardcoded it for now, to enable link to the next page. To implement it correct way we should have collection with metadata, which will contain count of documents in other collection, then increment/decrement  counter on every create/delete of document

Reply:
Does firebase charge $$ if we just do a simple count(*) type query? If it just counts as 1 document query I think that would be fine for now since the launch time is soon. There would be a lot of touchpoints if we need every insert/delete to +1 or -1 (which itself is 1 DB update, unless it can be done in the same update query). You also might also get into a concurrency issue...I don't know how well firebase handles atomicity...but in general nosql doesn't :)

Reply:
@fodaley AFAIK Firebase doesn't have count query. You have to download all documents (each one counts as one read) and count it by itself, or most common, use something like https://firebase.google.com/docs/firestore/solutions/counters.